app/models/repaymentHistory/RepaymentHistoryUtils.scala

```diff
@@ -94,20 +100,20 @@ object RepaymentHistoryUtils {
   private def filterPayment(payment: Payment,
                             isAgent: Boolean,
                             reviewAndReconcileEnabled: Boolean
-                           )(implicit messages: Messages, dateservice: DateServiceInterface): Option[PaymentHistoryEntry] = {
+                           )(implicit messages: Messages, dateservice: DateServiceInterface): Either[Throwable, PaymentHistoryEntry] = {
 
     val hasCredit = payment.credit.isDefined
     val hasLot = payment.lot.isDefined && payment.lotItem.isDefined
 
     (hasCredit,  reviewAndReconcileEnabled, hasLot, payment.creditType) match {
-      case (true, _, _, Some(MfaCreditType))                                   => Some(mfaCreditEntry(payment, isAgent))
-      case (true, _, _, Some(CutOverCreditType))                               => Some(creditEntry(payment, isAgent))
-      case (true, true, _, Some(PoaOneReconciliationCredit))  => Some(creditEntry(payment, isAgent, true))
-      case (true, true, _, Some(PoaTwoReconciliationCredit))  => Some(creditEntry(payment, isAgent, true))
-      case (true, _, _, Some(BalancingChargeCreditType))                       => Some(creditEntry(payment, isAgent))
-      case (true, _, _, Some(RepaymentInterest))                               => Some(creditEntry(payment, isAgent))
-      case (false, _, true, Some(PaymentType))                                 => Some(paymentToHMRCEntry(payment, isAgent))
-      case (_, _, _, _)                                                        => None
+      case (true, _, _, Some(MfaCreditType))                                   => Right(mfaCreditEntry(payment, isAgent))
+      case (true, _, _, Some(CutOverCreditType))                               => creditEntry(payment, isAgent)
+      case (true, true, _, Some(PoaOneReconciliationCredit))  => creditEntry(payment, isAgent, true)
+      case (true, true, _, Some(PoaTwoReconciliationCredit))  => creditEntry(payment, isAgent, true)
+      case (true, _, _, Some(BalancingChargeCreditType))                       => creditEntry(payment, isAgent)
+      case (true, _, _, Some(RepaymentInterest))                               => creditEntry(payment, isAgent)
+      case (false, _, true, Some(PaymentType))                                 => Right(paymentToHMRCEntry(payment, isAgent))
+      case (_, _, _, _)                                                        => Left(MissingFieldException("Invalid Payment Data"))
```

Comment:
In case we are promoting this case to be an error,  I guess worth to log actual data / payment entity like on line 121;
but not necessarily we have to use prettyPrint:
`Logger("application").info("json:" + Json.prettyPrint(Json.toJson(payment))) `